Workflow/Workflow_FailoverFailback.ps1

```diff
@@ -0,0 +1,232 @@
+
+
+Remove-VM -DeletePermanently -VM $vm -Confirm:$false
+
+
+Write-Output ">>> Run 'Get-TintriVM' to refresh the tintri server $target_tintriserver.`n"
+
+Get-TintriVM -TintriServer $target_ts
+
+
+Write-Output ">>> Restore vm $vmname on tintri server $target_tintriserver.`n"
+
+Restore-TintriVM -Name $vmname -TintriServer $target_ts -DestinationDirectory $target_destination_directory -UseLatestSnapshot
+
+
+Write-Output ">>> Add vm $vmname on tintri server $target_tintriserver to inventory.`n"
+
+$target_vmfilepath = '[' + $target_ts.HostNameOrIp + '] ' + $target_destination_directory + '/' + $vmname + '.vmx'
+
+New-VM -VMFilePath $target_vmfilepath -VMHost $target_esx
+
+
+Write-Output ">>> Run 'Get-TintriVM' to refresh the tintri server $target_tintriserver.`n"
+
+Get-TintriVM -TintriServer $target_ts
+
+
+Write-Output ">>> Get the tintri vm $vmname on tintri server $target_tintriserver.`n"
+
+$target_vm = Get-TintriVM -Name $vmname -TintriServer $target_ts
+
+$target_vm
+
+
+Write-Output ">>> Get the datastore repl path for tintri server $target_tintriserver.`n"
+
+$target_rp = Get-TintriDatastoreReplPath -HostName $target_ts.ApplianceHostName
+
+$target_rp
+
+
+Write-Output ">>> Replicate vm $vmname from tintri server $target_tintriserver to $source_tintriserver.`n"
+
+$vmrepl = New-TintriVMReplConfiguration -VM $target_vm -DatastoreReplPath $target_rp
+
+while ( !$vmrepl ) {
+
+    $source_vmlist = Get-TintriVM -TintriServer $source_ts
+
+    $target_vmlist = Get-TintriVM -TintriServer $target_ts
+
+    $vmrepl = New-TintriVMReplConfiguration -VM $target_vm -DatastoreReplPath $target_rp
+    
+}
+
+$vmrepl
+
+
+Write-Output ">>> Make sure the replication is completed. Use 'Get-TintriVM' to refresh the VMStore.`n"
+
+do {
+
+    $vmlist = Get-TintriVM -TintriServer $target_ts
+
+    $latest_ss = (Get-TintriVM -Name $vmname -TintriServer $target_ts).Snapshot.Latest.Uuid.UuId
+
+    $latest_repl = (Get-TintriVM -Name $vmname -TintriServer $target_ts).Snapshot.LatestReplicated.Uuid.UuId
+    
+}
+while( $latest_ss -ne $latest_repl )
+
+
+Write-Output ">>> Remove the replication for vm $vmname on tintri server $target_tintriserver.`n"
+
+Remove-TintriVMReplConfiguration -VM $target_vm
+
+
+Write-Output ">>> Delete vm $vmname from tintri server $target_tintriserver.`n"
+
+$vm = Get-VM -Name $vmname
+
+Stop-VM -VM $vm -Confirm:$False
+
+Remove-VM -DeletePermanently -VM $vm -Confirm:$false
+
+
+Write-Output ">>> Run 'Get-TintriVM' to refresh the tintri server $source_tintriserver.`n"
+
+Get-TintriVM -TintriServer $source_ts
+
+
+Write-Output ">>> Restore vm $vmname on tintri server $source_tintriserver.`n"
+
+Restore-TintriVM -Name $vmname -TintriServer $source_ts -DestinationDirectory $source_destination_directory -UseLatestSnapshot
+
+
+Write-Output ">>> Add vm $vmname on tintri server $source_tintriserver to inventory.`n"
+
+$source_vmfilepath = '[' + $source_ts.HostNameOrIp + '] ' + $source_destination_directory + '/' + $vmname + '.vmx'

```

Comment:
Need to change path:
$source_datastore = Get-TintriHypervisorDatastore -TintriServer $source_ts

$source_vmfilepath = '[' + $source_datastore.DisplayName + '] ' + $source_destination_directory + '/' + $vmname + '.vmx'